### DISCUSSION TITLE
--affected and additional filters

### DISCUSSION THREAD

**Discussion by SilentImp** [vercel/turborepo#9076] [Help] [answered]
### Summary

Hi, 
I want to run command against affected apps and packages and one that depends on them.
If I understand correctly, `--affected` is alias for `--filter=[origin/main]`.
To also get dependent ones, I need to add `--filter=...[origin/main]`.
But in case I try to use them together `turbo test --filter=...[origin/main] --affected` I am getting error:  `ERROR  the argument '--filter <FILTER>' cannot be used with '--affected'`.

So, how to then I should do that? 
My only option is `turbo test --filter=...[origin/main] --filter=[origin/main]`?

**Comment by anthonyshew** [comment, marked as answer]
`--affected` will use your dependency graph, so you don't need to manually include dependents. If you B depends on A and A has changes, B will also be considered affected.

Regarding using filters with `--affected`, yes, this is a conscious limitation. You can work around it as you've described, using [`--affected`'s equivalent comparison](https://turbo.build/repo/docs/reference/run#--affected).

**Comment by ethanresnick** [comment]
> Regarding using filters with --affected, yes, this is a conscious limitation

@anthonyshew Can you say a bit more about the rationale for this? I think I have a use case where combining `--affected` with other filters would be helpful, and where it's hard to replace `--affected` with its `--filter` equivalent because we're relying on the logic for base branch inference in Github Actions
